Sources/Sentry/include/SentryClient.h

```diff
@@ -47,19 +47,23 @@ SENTRY_NO_INIT
 
 @property(nonatomic, strong) SentryOptions *options;
 
-
 /**
- * This block can be used to modify the event before it will be serialized and sent
+ * Defines the sample rate of SentryClient, should be a float between 0.0 and 1.0
+ * Setting this property sets shouldSendEvent callback and applies a random event sampler.
  */
-@property(nonatomic, copy) SentryBeforeSerializeEvent _Nullable beforeSerializeEvent;
+@property(nonatomic) float sampleRate;
 
 /**
  * This will be filled on every startup with a dictionary with extra, tags, user which will be used
  * when sending the crashreport
  */
 @property(nonatomic, strong) NSDictionary<NSString *, id> *_Nullable lastContext;
 
-@property(nonatomic, strong) SentryTransport* transport;
+/**
+ * This block can be used to prevent the event from being sent.
+ * @return BOOL
+ */
+@property(nonatomic, copy) SentryShouldSendEvent _Nullable shouldSendEvent;
```

Comment:
We don't need this anymore, if `beforeSend` returns `nil` it should not send the event, we probably need to do the sample rate differently and not in this callback. More in `prepareEvent`